Bugfix/sling 5947

buuhuu opened the pull request:
I implementing the methods throwing UnsupportedOperationException when the MockSlingHttpServletRequest is used with the SlingRequestProcessor. 

Unfortunately there are some older commits included in this PR, which actually don't change anything. Hope thats not a problem.

buuhuu committed:
Merge pull request #1 from apache/trunk

Sync Fork

Commit:
SLING-5070: check the cacheSize before registering the SlingServletResolverMBean and register it only when there is a cache present.

Commit:
SLING-5068:
 - used a ThreadLocal for the activeResource
 - store a reference to the resources path for getting it from the shared ResourceResolver
 - in case the activeResource of the current thread is null or the ResourceResolver is not alive anymore we obtain the activeResource from the shared ResourceResolver

Commit:
merged upstream/trunk

Commit:
SLING-5947:
 - implemented getRequestURL, getRequestURI, getPathInfo, getAuthType, getServletPath

Commit:
SLING-5947:
 - reverted formatting changes
 - reverted to version 1.01-SNAPSHOT

Commit:
SLING-5947:
 - reverted formatting changes

kwin commented on bundles/extensions/servlet-helpers/src/main/java/org/apache/sling/servlethelpers/MockSlingHttpServletRequest.java:
```diff
@@ -632,56 +634,111 @@ public void setRemotePort(int remotePort) {
+        requestUrl.append(this.getScheme());
+        requestUrl.append("://");
+        requestUrl.append(getServerName());
+        requestUrl.append(':');
```
the port should only be included if it is not 80 for scheme=http or 443 for scheme=https

kwin commented on bundles/extensions/servlet-helpers/src/test/java/org/apache/sling/servlethelpers/MockSlingHttpServletRequestTest.java:
```diff
@@ -88,6 +88,44 @@ public void testSession() {
     }
 
```
A test for getRequestURL would be useful as well showing the default port behaviour

Commit:
SLING-5947:
 - don't add the port to the requestUrl in case its the default port for schema http or https
 - added test for calculating the requestUrl

buuhuu commented:
Resolved.
